dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/spirv/VisitorSpirvOutput.java

```diff
@@ -94,7 +94,8 @@ private Expression acceptAssertionValue(SpirvParser.AssertionValueContext ctx, b
             throw new ParsingException("Uninitialized location %s", name);
         }
         TerminalNode offset = ctx.ModeHeader_PositiveInteger();
-        int o = offset == null ? 0 : Integer.parseInt(offset.getText());
+        int byteWidth = base.getType().getBitWidth() / 8;
```

Comment:
It will not be 8 for non 64-bit values. You can get the actual bit width as TYPE_FACTORY.getMemorySizeInBytes(builder.getVariableType(name));

I would also add one-two tests with different width if we change it.

Reply:
Updated, I added an Int32 test.